src/circular_buffer.jl

```diff
@@ -9,6 +9,11 @@ type CircularBuffer{T} <: AbstractVector{T}
     (::Type{CircularBuffer{T}}){T}(capacity::Int) = new{T}(capacity, 1, T[])
 end
 
+function Base.empty!(cb::CircularBuffer)
+    cb.buffer = typeof(cb.buffer)[]
+    cb
+end
```

Comment:
I think you also need to reset `cb.first = 1`.  Is there a reason you didn't just call `empty!(cb.buffer)`?

Reply:
Also, could you add a quick test for this?

Reply:
Changes make. Test added. You're suggestion is much faster. Thanks for the help!